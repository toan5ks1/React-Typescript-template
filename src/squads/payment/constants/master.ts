// Payment
export enum OrderManagementCategory {
    Discount = "discount",
    Fee = "fee",
    LeavingReason = "leavingReason",
    Material = "material",
    Package = "package",
    PackageCourseMaterial = "packageCourseMaterial",
    PackageCourseFee = "packageCourseFee",
    PackageQuantityTypeMapping = "packageQuantityTypeMapping",
    ProductAccountingCategory = "productAccountingCategory",
    ProductCourse = "productCourse",
    ProductGrade = "productGrade",
    ProductLocation = "productLocation",
    ProductPrice = "productPrice",
    Tax = "tax",
    AccountingCategory = "accountingCategory",
    BillingRatio = "billingRatio",
    BillingSchedule = "billingSchedule",
    BillingSchedulePeriod = "billingSchedulePeriod",
}

// User
export enum UserSchoolGroupCategory {
    SchoolLevel = "schoolLevel",
    School = "school",
    SchoolCourse = "schoolCourse",
    SchoolLevelGrade = "schoolLevelGrade",
}

export enum UserTagGroupCategory {
    UserTag = "userTag",
}

export enum UserBankGroupCategory {
    Bank = "bank",
    BankBranch = "bankBranch",
}

export enum TimesheetCategory {
    TimesheetConfig = "timesheetConfig",
}

// Architecture
export enum ArchitectureReserveCategory {
    Grade = "grades",
}
export enum ArchitectureCategory {
    Class = "class",
    Location = "location",
    LocationType = "locationType",
}

// Invoice
export enum InvoiceCategory {
    InvoiceSchedule = "invoiceSchedule",
}

export type MasterCategoryType =
    | OrderManagementCategory
    | UserSchoolGroupCategory
    | UserTagGroupCategory
    | UserBankGroupCategory
    | TimesheetCategory
    | ArchitectureReserveCategory
    | ArchitectureCategory
    | InvoiceCategory;
