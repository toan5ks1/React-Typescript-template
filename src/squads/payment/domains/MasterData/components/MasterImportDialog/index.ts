export * from "./MasterImportDialog";
export { default } from "./MasterImportDialog";
