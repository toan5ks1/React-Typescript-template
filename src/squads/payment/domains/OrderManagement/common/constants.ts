import { ArrayElement } from "src/squads/payment/types/common/array";
import { OrderFormValues } from "src/squads/payment/types/form/order-form-types";

export const FIRST_STUDENT_IDX = 0;

export const defaultProductFieldArrayItems: ArrayElement<
    OrderFormValues["students"]
>["productFieldArrayItems"] = [
    {
        product: null,
    },
];
