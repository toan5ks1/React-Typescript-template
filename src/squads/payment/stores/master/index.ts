export * from "./actions";
export { default } from "./reducer";
export * from "./selectors";
