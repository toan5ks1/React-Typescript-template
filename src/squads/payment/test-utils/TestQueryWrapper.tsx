import { PropsWithChildren } from "react";

import { QueryClientProvider, QueryClient } from "react-query";

const queryClient = new QueryClient();

const TestQueryWrapper = ({ children }: PropsWithChildren<{}>) => {
    return <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>;
};

export default TestQueryWrapper;
