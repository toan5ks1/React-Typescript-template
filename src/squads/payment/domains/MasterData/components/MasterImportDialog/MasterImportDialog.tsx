import { Entities } from "src/common/constants/enum";
import { MasterCategoryType } from "src/squads/payment/constants/master";

import DialogWithHeaderFooter from "src/components/Dialogs/DialogWithHeaderFooter";
import MasterImport from "src/squads/payment/domains/MasterData/components/MasterImport";

import { UseDialogReturn } from "src/squads/payment/hooks/useDialog";
import useResourceTranslate from "src/squads/payment/hooks/useResourceTranslate";

export interface MasterImportDialogProps extends Pick<UseDialogReturn, "open" | "onClose"> {
    category: MasterCategoryType;
}

export const MasterImportDialog = ({ open, onClose, category }: MasterImportDialogProps) => {
    const tMaster = useResourceTranslate(Entities.MASTERS);

    return (
        <DialogWithHeaderFooter
            data-testid="MasterImportDialog__root"
            open={open}
            onClose={onClose}
            title={tMaster("title.import")}
            maxWidth="md"
            minWidthBox="sm"
            footer="portal"
        >
            <MasterImport category={category} onClose={onClose} />
        </DialogWithHeaderFooter>
    );
};

export default MasterImportDialog;
