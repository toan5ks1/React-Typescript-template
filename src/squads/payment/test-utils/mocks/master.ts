import { Entities, MIMETypes } from "src/common/constants/enum";
import { OptionSelectType } from "src/common/constants/types";
import {
    ArchitectureCategory,
    ArchitectureReserveCategory,
    InvoiceCategory,
    OrderManagementCategory,
    UserSchoolGroupCategory,
} from "src/squads/payment/constants/master";
import { NsPaymentMasterImportService } from "src/squads/payment/service/payment/master-import-payment-service/types";

import { generateFiles } from "./file";

export const createMockMasterTypeOptions = (
    featuresGroup: "Architecture" | "Architecture_Reserved" | "User" | "OrderManagement" | "Invoice"
): OptionSelectType[] => {
    switch (featuresGroup) {
        case "Architecture":
            return [
                {
                    id: ArchitectureCategory.Location,
                    value: ArchitectureCategory.Location,
                    label: "Location",
                },
                {
                    id: ArchitectureCategory.LocationType,
                    value: ArchitectureCategory.LocationType,
                    label: "LocationType",
                },
                {
                    id: ArchitectureCategory.Class,
                    value: ArchitectureCategory.Class,
                    label: "Class",
                },
            ];
        case "Architecture_Reserved":
            return [
                {
                    id: ArchitectureReserveCategory.Grade,
                    value: ArchitectureReserveCategory.Grade,
                    label: "Grade",
                },
            ];
        case "User":
            return [
                {
                    id: UserSchoolGroupCategory.School,
                    value: UserSchoolGroupCategory.School,
                    label: "School",
                },
            ];
        case "Invoice":
            return [
                {
                    id: InvoiceCategory.InvoiceSchedule,
                    value: InvoiceCategory.InvoiceSchedule,
                    label: "InvoiceSchedule",
                },
            ];
        default:
            return [];
    }
};

export const createMockCSVFiles = (count: number = 1) => {
    const fileSchemas = Array.from(Array(count), (_, i) => i + 1).map((number) => ({
        size: 300 * number,
        fileProperty: {
            type: MIMETypes.CSV,
        },
    }));

    return generateFiles(fileSchemas);
};

export const createMockMasterImportFile = (id: string = "1", isImporting: boolean = false) => {
    const file = createMockCSVFiles()[0];
    return {
        id: `${Entities.MASTERS}_${id}}`,
        file,
        isImporting,
    };
};

//services

export const createMockMasterImportPayload =
    (): NsPaymentMasterImportService.ImportMasterDataPayload => {
        const file = createMockCSVFiles()[0];

        return {
            file,
            category: OrderManagementCategory.AccountingCategory,
        };
    };

export const createMockMasterImportPayloadBob =
    (): NsPaymentMasterImportService.ImportMasterRequest["payload"] => {
        const file = createMockCSVFiles()[0];

        return {
            file,
            category: ArchitectureCategory.Location,
        };
    };
