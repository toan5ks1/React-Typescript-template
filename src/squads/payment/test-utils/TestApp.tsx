import { PropsWithChildren } from "react";

import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import { createEmptyRootState } from "src/squads/payment/test-utils/mocks/root-state";

import ThemeProvider from "src/squads/payment/contexts/ThemeProvider";
import TranslationProvider from "src/squads/payment/contexts/TranslationProvider";

const mockRootState = createEmptyRootState();

const TestApp = ({ children }: PropsWithChildren<{}>) => {
    const store = configureStore()(mockRootState);

    return (
        <Provider store={store}>
            <BrowserRouter>
                <TranslationProvider>
                    <ThemeProvider>{children}</ThemeProvider>
                </TranslationProvider>
            </BrowserRouter>
        </Provider>
    );
};

export default TestApp;
