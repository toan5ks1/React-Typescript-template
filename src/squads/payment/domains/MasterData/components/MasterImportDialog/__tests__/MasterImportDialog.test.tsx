import { OrderManagementCategory } from "src/squads/payment/constants/master";

import { MasterImportDialog } from "../MasterImportDialog";

import { render, RenderResult } from "@testing-library/react";
import TestApp from "src/squads/payment/test-utils/TestApp";
import TestQueryWrapper from "src/squads/payment/test-utils/TestQueryWrapper";
import TestThemeProvider from "src/squads/payment/test-utils/TestThemeProvider";

describe("<MasterImportDialog />", () => {
    let wrapper: RenderResult;
    const onClose = jest.fn();
    beforeEach(() => {
        (onClose as jest.Mock).mockReturnValue(onClose);

        wrapper = render(
            <TestApp>
                <TestThemeProvider>
                    <TestQueryWrapper>
                        <MasterImportDialog
                            category={OrderManagementCategory.AccountingCategory}
                            open={true}
                            onClose={onClose}
                        />
                    </TestQueryWrapper>
                </TestThemeProvider>
            </TestApp>
        );
    });

    it("should match snapshot", () => {
        expect(wrapper.container).toMatchSnapshot();
    });

    it("should render UI correctly", () => {
        expect(wrapper.getByTestId("MasterImportDialog__root")).toBeInTheDocument();
        expect(wrapper.getByTestId("DialogWithHeaderFooter__dialogTitle")).toBeInTheDocument();
        expect(wrapper.getByTestId("FooterDialogConfirm__buttonSave")).toBeInTheDocument();
        expect(wrapper.getByTestId("FooterDialogConfirm__buttonClose")).toBeInTheDocument();
    });
});
