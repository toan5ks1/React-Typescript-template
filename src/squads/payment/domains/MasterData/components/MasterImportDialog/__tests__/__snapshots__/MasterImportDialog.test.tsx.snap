// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MasterImportDialog /> should match snapshot 1`] = `
<div
  aria-hidden="true"
/>
`;
